### PR TITLE
(DOCSP-26538): Kotlin @PersistedName annotation documentation

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Add documentation for `@PersistedName` annotation 

### Jira

- https://jira.mongodb.org/browse/DOCSP-26538

### Staged Changes

- new page: [Remap a Property - Kotlin SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-26538-persisted/sdk/kotlin/realm-database/schemas/remap/)
- updated page: [Define a Realm Object Model - Kotlin SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-26538-persisted/sdk/kotlin/realm-database/define-realm-object-model/#map-a-property-to-a-different-name)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/realm-database/schemas/remap.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- To change a field name in Kotlin without forcing a :ref:`migration <kotlin-migrations>`.
+
+To map a different property name in your code than is stored in
+Realm Database, use the `@PersistedName <{+kotlin-local-prefix+}io.realm.kotlin.types.annotations/-persisted-name/index.html>`__
+annotation in the :ref:`object schema <kotlin-object-schema>`:
+
+.. literalinclude:: /examples/generated/kotlin/SchemaTest.snippet.persisted-name.kt
+   :language: kotlin
+
+Note that migrations must use the internal name when creating classes and
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database/schemas/remap.txt:27]: I don't think "internal name" inherently has meaning in this context. I could see that being interpreted either as the name that the project codebase uses, or the name that realm stores. If you borrow from the Swift paragraph above, you might want to reword this to language similar to that - i.e. "public property name used in the code throughout the project" and "private property name that is persisted to realm". (It's not concise and could probably be wordsmithed, but I think clarity here is more important than brevity.)

File: source/sdk/kotlin/realm-database/define-realm-object-model.txt [hunk 2]
```diff
@@ -124,6 +124,19 @@ For more information, refer to :ref:`Primary Keys <kotlin-primary-keys>`.
   :language: kotlin
   :emphasize-lines: 2-3
 
+Map a Property to a Different Name 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To indicate a property be persisted under a different name,
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database/define-realm-object-model.txt:130]: This is a very subjective nit, but "To indicate a property be persisted under a different name" reads as a bit indirect and awkward to me. Maybe we could be more direct here?
```suggestion
To persist a field under a different property name, 
```

File: source/sdk/kotlin/realm-database/define-realm-object-model.txt [hunk 3]
```diff
@@ -124,6 +124,19 @@ For more information, refer to :ref:`Primary Keys <kotlin-primary-keys>`.
   :language: kotlin
   :emphasize-lines: 2-3
 
+Map a Property to a Different Name 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To indicate a property be persisted under a different name, 
+use the ``@PersistedName`` annotation. 
+This is useful when open a Realm across different bindings
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database/define-realm-object-model.txt:132]: Alternately, "when opening"?
```suggestion
This is useful when you open a Realm across different bindings
```

**Review by dacharyc — APPROVED** [review]
There's one missing word that should be fixed, and a handful of nits/suggestions for wording changes or clarifications, but overall LGTM.

File: source/sdk/kotlin/realm-database/schemas/remap.txt (hunk 1 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database/schemas/remap.txt]: Agreed! Updated for clarity
